include/extractor/files.hpp

```diff
@@ -0,0 +1,160 @@
+#ifndef OSRM_EXTRACTOR_FILES_HPP
+#define OSRM_EXTRACTOR_FILES_HPP
+
+#include "extractor/guidance/turn_lane_types.hpp"
+#include "extractor/seralization.hpp"
+
+#include "util/coordinate.hpp"
+#include "util/packed_vector.hpp"
+#include "util/serialization.hpp"
+
+#include <boost/assert.hpp>
+
+namespace osrm
+{
+namespace extractor
+{
+namespace files
+{
+
+// reads .osrm.nodes
+template <bool UseShareMemory>
+inline void readNodes(const boost::filesystem::path &path,
+                      typename util::ShM<util::Coordinate, UseShareMemory>::vector &coordinates,
+                      util::detail::PackedVector<OSMNodeID, UseShareMemory> &osm_node_ids)
+{
+    const auto fingerprint = storage::io::FileReader::VerifyFingerprint;
+    storage::io::FileReader reader{path, fingerprint};
+
+    reader.DeserializeVector(coordinates);
+    util::serialization::read(reader, osm_node_ids);
+}
+
+// writes .osrm.nodes
+template <bool UseShareMemory>
+inline void
+writeNodes(const boost::filesystem::path &path,
+           const typename util::ShM<util::Coordinate, UseShareMemory>::vector &coordinates,
+           const util::detail::PackedVector<OSMNodeID, UseShareMemory> &osm_node_ids)
+{
+    const auto fingerprint = storage::io::FileWriter::GenerateFingerprint;
+    storage::io::FileWriter writer{path, fingerprint};
+
+    writer.SerializeVector(coordinates);
```

Comment:
From outside it looks strange to have both `writer.SerializeVector` and `util::serialization::write(writer, ...)` Would it make sense to unify interface as `util::serialization::write(writer, vector)` ?

Or even with syntactic sugar `writer << vector` ?